snewpdag/plugins/DistCalc1.py

```diff
@@ -35,18 +39,34 @@ class DistCalc1(Node):
                'JUNO, NO': [128.54796152661447, 20.633251789516653, 0.16051014379753317], \
                'JUNO, IO': [135.26455501942974, 18.91887993385422, 0.13986576107197238]}
     
-    def __init__(self, detector, **kwargs):
+    def __init__(self, detector, in_field, out_field, **kwargs):
         self.detector = detector
+        self.in_field = in_field
+        self.out_field = out_field
         super().__init__(**kwargs)
     
     def dist_calc1(self, data):
-
-        bg = np.mean(data[0:1000]) #using first 1000 bins to find background
-        N50 = np.sum(data[1000:1500]-bg) #correct for background
+        bg = np.mean(data[self.in_field][0:1000]) #using first 1000 bins to find background
+        N50 = np.sum(data[self.in_field][1000:1500]-bg) #correct for background
```

Comment:
N50 should sum over 50 bins of 1ms, not over 500 (?)

Reply:
We summed over 50 bins because we used a bin-size of 0.1 ms, which is an assumption of the data we referenced from the TimeDistDiff plugin.